src/main/java/seedu/address/model/person/ModuleRoleMap.java

```diff
@@ -0,0 +1,232 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Stream;
+
+/**
+ * Represents the mapping between module and role type in the roles
+ * taken by a Person in NUS.
+ */
+public class ModuleRoleMap {
+    /* Constraints message for role type. */
+    public static final String MESSAGE_CONSTRAINTS =
+            "Role consists of two parts, module code and role type.\n"
+            + "Role type can be blank, and or any of the following values(case insensitive) for each role type:\n"
+            + "1. Student: student or leave blank\n"
+            + "2. Tutor: tutor or ta\n"
+            + "3. Professor: professor or prof";
+
+    /*
+     * Input keyword must match one of the keyword provided in the regex.
+     * Note that the pattern is only matching lower case, so the other methods
+     * need to convert user input to lower case first before validate the input
+     */
+    public static final String VALIDATION_REGEX = "^$|student|tutor|ta|professor|prof";
+
+    private final HashMap<ModuleCode, RoleType> roles;
+
+    /**
+     * Default constructor for a {@code ModuleRoleMap}.
+     * This is for creating roles for a Person based on the module code and role types
+     * extracted from an add command.
+     *
+     * @param moduleCodes Array of modules codes
+     * @param roleTypes Array of role type String
+     */
+    public ModuleRoleMap(ModuleCode[] moduleCodes, RoleType[] roleTypes) {
+        requireNonNull(moduleCodes);
+        requireNonNull(roleTypes);
+
+        if (moduleCodes.length != roleTypes.length) {
+            throw new IllegalArgumentException(
+                    "Module code array's length is not consistent with that of role type array.");
+        }
+
+        checkArgument(isValidModuleCodeArray(moduleCodes), ModuleCode.MESSAGE_CONSTRAINTS);
+        checkArgument(isValidRoleTypeArray(roleTypes), MESSAGE_CONSTRAINTS);
+
+        HashMap<ModuleCode, RoleType> newRoles = new HashMap<>();
+        for (int i = 0; i < moduleCodes.length; i++) {
+            newRoles.put(moduleCodes[i], roleTypes[i]);
+        }
+        this.roles = newRoles;
+    }
+
+    /**
+     * Constructor for a {@code ModuleRoleMap}.
+     * Copy of the hashmap is performed to make the class immutable.
+     *
+     * @param roles new roles which are assigned to the Person
+     */
+    public ModuleRoleMap(HashMap<ModuleCode, RoleType> roles) {
+        requireNonNull(roles);
+        this.roles = new HashMap<>(roles);
+    }
+
+    /**
+     * Returns true if the given role type String array contains all
+     * valid role type keywords.
+     *
+     * @param roleTypes Array of role type keywords
+     * @return whether the array contains all valid keywords
+     */
+    public static boolean areValidRoleTypeKeywords(String[] roleTypes) {
```

Comment:
I think it makes more sense to move this to RoleType class.
Same for roleTypeStringToEnum